### DISCUSSION TITLE
showCreateRelationButton() not including Related Model Field

### DISCUSSION THREAD

**Discussion by jjjrmy** [laravel/nova-issues#4402] [Ideas & Feature Requests]
- Nova Version: 4.8.1

**/nova/Input.php**
```php
BelongsToMany::('Options')->showCreateRelationButton()
```
`/nova/resources/inputs/<id>/attach/options?viaRelationship=options&polymorphic=0`

When I click the Create button on the Relationship, I want to be able to know the ID of the requesting Resource for use in the Modal: to get the Input's ID.
It should be possible in the `NovaRequest $request` but I am not seeing anything.

**Comment by crynobone** [comment]
Duplicate of https://github.com/laravel/nova-issues/issues/3335

**Comment by frtess** [comment]
I'm also wondering about this. How can I specify default values, filter dropdowns or completely customize the "Attach" screen. If I can override the "Attach resource" on the parent resource, I'll just make a custom [resource]AttachTool, but can I do this ?

I'm new to Nova but a seasoned developer. I find Nova incredibly powerful, for basic crud (including Laravel relationships), but it seems to lack in term of API documentation. To achieve their moto "Making [us] the most productive developers", complete docs on the extents of possible customizations are needed.

**Comment by jjjrmy** [comment]
Here is an example of why this is necessary:

A Form `hasMany` Steps, and also `belongsTo` it's First Step.
So when Updating (and Creating) the Form, it's not possible to create it's First Step because you can't set the `belongsTo` on the Step that's being newly created - even though the Form model already exists.

The issue is even more complex when Creating, because the Form model doesn't exist yet, but not sure if there is a solution for that.
